app/controllers/facility_journals_controller.rb

```diff
@@ -26,23 +27,31 @@ def index
   end
 
   # GET /facilities/journals/new
-  def new_with_search
+  def new
     raise ActiveRecord::RecordNotFound if current_facility.cross_facility?
 
-    set_default_variables
-    @layout = "two_column_head"
+    order_details = OrderDetail.for_facility(current_facility).need_journal
+    @search_form = TransactionSearch::SearchForm.new(params[:search])
+    @search = TransactionSearch::Searcher.search(order_details, @search_form)
+    @date_range_field = @search_form.date_params[:field]
+    @order_details = @search.order_details
 
-    @after_search_hook = lambda do
-      @valid_order_details, @invalid_order_details = ValidatorFactory.partition_valid_order_details(@order_details.unexpired_account)
-      @invalid_order_details += @order_details.expired_account
+    set_earliest_journal_date
+
+    unless current_facility.has_pending_journals?
+      @order_detail_action = :create
+      @action_date_field = { journal_date: @earliest_journal_date }
+    end
 
-      respond_to do |format|
-        format.csv do
-          # used for "Export as CSV" link for order details with expired accounts
-          @order_details = @invalid_order_details
-        end
-        format.any {}
+    @valid_order_details, @invalid_order_details = ValidatorFactory.partition_valid_order_details(@order_details.unexpired_account)
```

Comment:
Wow, we're doing a lot in this action. Would there be in value in extracting some smaller methods?

Reply:
I agree there's a lot going on here. I'm a little leery of pulling out smaller methods though, because a lot of what's happening here is setting instance variables that are used in the various views, and I think if we hide those in other methods the code becomes a lot more confusing. I'm willing to be convinced otherwise though.